Code/Tools/SceneAPI/SceneData/ManifestBase/SceneNodeSelectionList.cpp

```diff
@@ -23,103 +23,59 @@ namespace AZ
                 return m_selectedNodes.size();
             }
 
-            const AZStd::string& SceneNodeSelectionList::GetSelectedNode(size_t index) const
+            void SceneNodeSelectionList::AddSelectedNode(const AZStd::string& name)
             {
-                AZ_Assert(index < m_selectedNodes.size(), "Invalid index %i for selected node in mesh group.", index);
-                return m_selectedNodes[index];
+                m_unselectedNodes.erase(name);
+                m_selectedNodes.emplace(name);
```

Comment:
If there is an existing unselected node, you can technically [extract](https://en.cppreference.com/w/cpp/container/unordered_set/extract) it from the unselected node set and then [insert](https://en.cppreference.com/w/cpp/container/unordered_set/insert) it into the selected node set for better performance gains.
```suggestion
                if (auto extractedNodeHandle = m_unselectedNodes.extract(name); extractedNodeHandle)
                {
                    m_selectedNodes.insert(extractedNodeHandle);
                }
                else
                {
                    m_selectedNodes.emplace(name);
                }
```

Reply:
ooh, good point!